iotfunctions/pipeline.py

```diff
@@ -2685,10 +2685,10 @@ def check_data_items_type(self, df, items):
                         # 'object'.
                         # Conversion via combination of where() and mask() is 60% faster than an approach with apply()
                         try:
-                            if isinstance(df_column.values[0], str):
-                                cond = df_column == 'True'
+                            if all( isinstance(val, str) or val is None for val in df_column.values):
```

Comment:
@singhshraddha One more suggestion: When you exchange 'val is None' by 'pd.isna(val)' your solution gets even more flexible.